CalendarApp/GoogleCalendarService/Manager/EventPeriodicityHolder.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Google.Apis.Calendar.v3.Data;
+
+namespace GoogleCalendarService.Manager
+{
+    public class EventPeriodicityHolder
+    {
+        public Event RealEvent;
+        public string Frequency;
+        public List<DayOfWeek> DayOfWeeksList;
+        public List<Event> WeekEventList;
+        public int Count;
+        public FrequencyType FreqType;
+        public DateTime UntilDateTime;
+        public List<Event> EventList;
+
+        public EventPeriodicityHolder(Event eventForAdd)
+        {
+            RealEvent = eventForAdd;
+            if (eventForAdd.Recurrence == null)
+            {
+                FreqType = FrequencyType.COUNT;
+                Count = 1;
+                Frequency = "DAILY";
+            }
+            else
+            {
+                string pattern = @":FREQ=(?<FREQ>\w+);(COUNT=(?<COUNT>\w+);)?(COUNT=(?<COUNT>\w+))?(UNTIL=(?<UNTIL>\d+\w+\d+\w+);)?(UNTIL=(?<UNTIL>\d+\w+\d+\w+))?(BYDAY=(?<BYDAY>[\w,]+))?";
+                Regex regex = new Regex(pattern);
+                Match match = regex.Match(eventForAdd.Recurrence[0]);
+
+                Frequency = match.Groups["FREQ"].Value;
+                if(match.Groups["BYDAY"].Value != "")
+                    SetDayOfWeeksList(match.Groups["BYDAY"].Value);
+                if (match.Groups["COUNT"].Value != "")
+                {
+                    FreqType = FrequencyType.COUNT;
+                    Count = Convert.ToInt32(match.Groups["COUNT"].Value);
+                }
+                if (match.Groups["UNTIL"].Value != "")
+                {
+                    FreqType = FrequencyType.UNTIL;
+                    UntilDateTime = GetUntilDateTime(match.Groups["UNTIL"].Value);
+                }
+            }
+
+            if(Frequency == "DAILY")
+                EventList = EventsListBuilder.StartConstruction().WithDailyEvents(this).Build();
+            else if(Frequency == "WEEKLY")
+                EventList = EventsListBuilder.StartConstruction().WithWeeklyEvents(this).Build();
+            else if (Frequency == "MONTHLY")
+                EventList = EventsListBuilder.StartConstruction().WithMonthlyEvents(this).Build();
+            else if (Frequency == "YEARLY")
+                EventList = EventsListBuilder.StartConstruction().WithYearlyEvents(this).Build();
```

Comment:
Frequency values can be extracted into enum or FrequencyTypes class that would contain constants per each frequency type